bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterHandler.java

```diff
@@ -0,0 +1,165 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.meater.internal.handler;
+
+import static org.openhab.binding.meater.internal.MeaterBindingConstants.*;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meater.internal.MeaterConfiguration;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO.Cook;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO.Device;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ * The {@link MeaterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterHandler extends BaseThingHandler {
+
+    private MeaterConfiguration config;
+    private TimeZoneProvider timeZoneProvider;
+
+    public MeaterHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        config = getConfigAs(MeaterConfiguration.class);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            update();
+        });
+    }
+
+    public void update() {
+        Device meaterProbe = getMeaterProbe();
+        if (meaterProbe != null) {
+            update(meaterProbe);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error.description");
+        }
+    }
+
+    private @Nullable Device getMeaterProbe() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            MeaterBridgeHandler bridgeHandler = (MeaterBridgeHandler) bridge.getHandler();
+            if (bridgeHandler != null) {
```

Comment:
You should also check that the bridge status is ONLINE. You return null if not.

Reply:
Fixed!